website/docs/concepts/router.md

```diff
@@ -289,6 +286,12 @@ pub fn nav_items() -> Html {
 }
 ```
 
+:::tip 
+If your component only needs to set the route without listening to the changes, instead of the `use_history`
+hook, `BrowserHistory::default()` can be used to acquire the global history instance. The latter also works in agent
+environments.
+:::
+
```

Comment:
I believe this is not recommended https://discordapp.com/channels/701068342760570933/701068343431528490/914461635786203136

Reply:
as @futursolo then himself added, the reason why it's not recommended:
> I wouldn't recommend BrowserHistory::default() because it does not cover other history kinds (which might come relatively soon). 

soon ™️ , but other history kinds won't be included in 0.19. So this will be the only way to do only set history in a component. We can change this part later.

Reply:
ok

Reply:
My personal opinion on this matter:

### History in agents

I think advising that BrowserHistory is available in browser is not 100% accurate.
History API is not available for “threaded” agents. You can only use it in context and job agents.

IMHO, if you need to navigate user away upon completion of an agent task, it’s better to send back the task result as a message and navigate in the component as the user might have already navigated away during the time your agent is completing the task.

### use_history and additional rendering

Just my 2 cents, but from my experience with React, it’s almost always the case that you cannot eliminate all unnecessary rendering from function components. I think it's just a tradeoff to make rendering automatic in function components.

I think for documentation, advertising good paradigm (use hooks) is more important here than how to hack to avoid a couple extra rendering.

(Unless `use_history` breaks your function component. In which case please let me know and I will submit a PR to fix.)

### Contexts for agents

It’s true that agents cannot access contexts easily but I think that’s a different issue and I might submit a PR to improve it.

Reply:
@futursolo same experience with react, you must be doing something wrong if a re render breaks things

Reply:
@futursolo @voidpumpkin thanks. Those are some tips and guidelines I didn't know!